src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -25,74 +25,39 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
   id: string;
 };
 
-const checkboxIconNameMap = {
-  indeterminate: 'checkbox-indeterminate',
-  true: 'checkbox-checked',
-  false: 'checkbox-unchecked',
-};
-const CheckboxSvg = ({
-  checked,
-  disabled,
-}: {
-  checked: boolean | 'indeterminate';
-  disabled?: boolean;
-}) => {
-  const iconClassName = clsx(
-    styles['checkbox__icon'],
-    disabled && styles['checkbox__icon--disabled'],
-  );
-  return (
-    <Icon
-      className={iconClassName}
-      name={
-        checkboxIconNameMap[`${checked}`] as
-          | 'checkbox-indeterminate'
-          | 'checkbox-checked'
-          | 'checkbox-unchecked'
-      }
-      purpose="decorative"
-      size="1.5rem"
-    />
-  );
-};
-
 /**
  * Checkbox input element, exported for greater flexibility.
  * You must provide an `id` prop and connect it to a visible label.
  */
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked = false, className, disabled, ...other }, ref) => {
-  // Make indeterminate checkbox visually match the colors of a
-  // checked state, but announce itself as "mixed" to screen readers
-  const checkedProps =
-    checked === 'indeterminate'
-      ? {
-          'aria-checked': 'mixed' as const,
-          checked: true,
-        }
-      : {
-          checked,
-        };
+>(({ checked, className, disabled, ...other }, ref) => {
+  const forwardedRef = useForwardedRef(ref);
+
+  // If the checkbox is "indeterminate", set that property on the node. This can only be done in JS
```

Comment:
Strange, but also auto handles `aria-checked=mixed`

Reply:
Yep.

Also, see https://github.com/facebook/react/issues/1798 for discussion on why React won't create an `indeterminate` prop (spoiler: because there isn't a DOM attribute, and adding one wouldn't play nice with SSR I guess).

Reply:
Actually, I'm going to revert doing this the "right" way. Want to handle that in a separate commit/PR.